diff --git a/src/rules.rs b/src/rules.rs
index 51bb96d..85d6ae6 100644
--- a/src/rules.rs
+++ b/src/rules.rs
@@ -58,7 +58,7 @@ pub(crate) fn spacing() -> SpacingDsl {
         .inside(NODE_LIST).before(T!["]"]).when(inline_with_attr_set).no_space()
         .test("[]", "[ ]")
         .inside(NODE_LIST).between(T!["["], T!["]"]).single_space_or_optional_newline()
-        .inside(NODE_LIST).between(VALUES, VALUES).single_space_or_newline()
+        .inside(NODE_LIST).between(VALUES, VALUES).single_space_or_optional_newline()
         .inside(NODE_LIST).between(VALUES, TOKEN_COMMENT).single_space_or_optional_newline()
         .inside(NODE_LIST).between(TOKEN_COMMENT, VALUES).single_space_or_newline()
 
